### DISCUSSION TITLE
Integrate OAM concepts with IDE and Dashboard

### DISCUSSION THREAD

**Discussion by lyzhang1999** [nocalhost/nocalhost#1132] [Ideas]
Currently, the KubeVela integration with Nocalhost can be done via the [`nocalhost` trait](https://github.com/oam-dev/kubevela/blob/3f2e76c75a1919bcc0fd8d3f7c776c6bcb48ac3c/docs/examples/nocalhost/app-bookinfo.yaml#L13). Here is a [live demo](https://www.bilibili.com/video/BV1mS4y1R7Ez).

## Problem with IDE

But after the KubeVela app is deployed, its structure looks like following in IDE:

![20211208153244](https://user-images.githubusercontent.com/920884/145167295-c0e02535-edcc-43a0-b42b-dd095977fe86.jpg)

To make the integration more KubeVela native, we should expose OAM concepts to users, e.g. Component/Trait. 

## Problem with Dashboard

The dashboard will display the application deployments. But it doesn't support *OAM-type* application:

![20211208153601](https://user-images.githubusercontent.com/920884/145167753-8cfae1cb-7548-4847-86eb-c90e5bd53714.jpg)

To make the integration more KubeVela native, we can add a OAM type.

proposal from @hongchaodeng

**Comment by FogDong** [comment]
![image](https://user-images.githubusercontent.com/15977536/148020869-a0e8c617-aefd-4817-a2a5-da69f19eff98.png)



After deploying a KubeVela Application with nocalhost trait, we can see the application under the namespace, which is awesome!

But under the application, we still have the regular workloads like deployment, maybe we can list `component` and `trait` if it's a KubeVela Application.

WDYT? @lyzhang1999

**Reply by lyzhang1999** [reply]
I think it's a good idea, Also the specific grouping information(component, trait) is only displayed when the application is of type KubeVela, which you mentioned is very important.
